src/main/java/io/divolte/server/Server.java

```diff
@@ -135,7 +204,9 @@ public void shutdown() {
         }
 
         logger.info("Stopping thread pools.");
-        processingPool.stop();
+        // Stop the mappings before the sinks to ensure work in progress doesn't get stranded.
+        mappingProcessors.values().forEach(ProcessingPool::stop);
+        sinks.values().forEach(ProcessingPool::stop);
 
         logger.info("Closing HDFS filesystem connection.");
```

Comment:
@krisgeus The sinks close their files, but not the HDFS client itself.

Previously the (single) processor pool closed down the client; this was moved into the server shutdown sequence. (Right now we have multiple processor pools, although we're going to try to get it back to a single one.)